### PR TITLE
fix(node): fix bug in reputation synchronization

### PR DESCRIPTION
**Pull request by lrubiorod**
Fixed a bug produced by updating active reputation system before than expected

### CODE REVIEW

File: validations/src/validations.rs [hunk 1]
```diff
@@ -724,7 +724,7 @@ pub fn validate_block(
     utxo_set: &UnspentOutputsPool,
     data_request_pool: &DataRequestPool,
     vrf: &mut VrfCtx,
-    rep_eng: &ReputationEngine,
```
Comment (tmpolaczyk) [inline validations/src/validations.rs]: The `validate_block` function should not mutate anything, we need to find another place to update the reputation engine.

Comment (lrubiorod) [reply validations/src/validations.rs]: The other option could be to put poe validation out of validate_block function, and update reputation engine before to call it

Comment (tmpolaczyk) [reply validations/src/validations.rs]: (1/2) But why is this update needed? Isn't enough to update the reputation engine...

File: node/src/actors/chain_manager/handlers.rs [hunk 2]
```diff
@@ -135,13 +129,14 @@ impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
...
+                        let previous_epoch = msg.checkpoint - 1;
                         log::warn!(
                             "There was no valid block candidate to consolidate for epoch {}",
-                            msg.checkpoint - 1
+                            previous_epoch
                         );
 
                         // Update ActiveReputationSet in case of epochs without blocks
-                        if let Err(e) = rep_engine.ars.update(vec![], expected_ars_epoch) {
+                        if let Err(e) = rep_engine.ars.update(vec![], previous_epoch) {
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/handlers.rs]: (2/2) ...here

Comment (lrubiorod) [reply node/src/actors/chain_manager/handlers.rs]: the problem are empty epochs while we are consolidating old blocks. For example:
We have a very few probability to mine, and there are only one miner now. After some empty epochs, the reputation engine is updated and now that miner has 100% of the probability and mine a block. If later other node has to synchronize that chain, if we dont update the reputation engine before to check the poe, this node keeps the few probability until it consolidate a new block, so the PoE that the other node show (f555... for example) is not valid for it, and reject a valid block thinking that it is not

Comment (tmpolaczyk) [reply node/src/actors/chain_manager/handlers.rs]: So it is a problem while synchronizing? Then we can handle it in `AddBlocks`

Comment (lrubiorod) [reply node/src/actors/chain_manager/handlers.rs]: :thinking: good point, I am going to try

**Pull request merged**